src/transactions/feed/TransactionFeedV2.test.tsx

```diff
@@ -0,0 +1,217 @@
+    showApprovalTxsInHomefeed: [NetworkId['celo-alfajores']],
+  })
+  jest.mocked(getDynamicConfigParams).mockReturnValue({
+    jumpstartContracts: {
+      ['celo-alfajores']: { contractAddress: '0x7bf3fefe9881127553d23a8cd225a2c2442c438c' },
+    },
+  })
+})
+
+describe('TransactionFeed', () => {
+  mockFetch.mockResponse(typedResponse({}))
+  it('renders correctly when there is a response', async () => {
+    mockFetch.mockResponse(typedResponse({ transactions: [mockTransaction()] }))
+    const { store, ...tree } = renderScreen()
+
+    await waitFor(() => expect(tree.getByTestId('TransactionList').props.data.length).toBe(1))
+    expect(tree.queryByTestId('NoActivity/loading')).toBeNull()
+    expect(tree.queryByTestId('NoActivity/error')).toBeNull()
+    expect(mockFetch).toHaveBeenCalledTimes(1)
+    expect(tree.getAllByTestId('TransferFeedItem').length).toBe(1)
+    expect(
+      within(tree.getByTestId('TransferFeedItem')).getByTestId('TransferFeedItem/title')
+    ).toHaveTextContent(
+      'feedItemReceivedTitle, {"displayName":"feedItemAddress, {\\"address\\":\\"0xd683...ea33\\"}"}'
+    )
+  })
+
+  it("doesn't render transfers for tokens that we don't know about", async () => {
+    mockFetch.mockResponse(typedResponse({ transactions: [mockTransaction()] }))
+    const { store, ...tree } = renderScreen()
+
+    await waitFor(() => tree.getByTestId('TransactionList'))
+    const items = tree.getAllByTestId('TransferFeedItem/title')
+    expect(items.length).toBe(1)
+  })
+
+  it('renders the loading indicator while it loads', async () => {
+    const tree = renderScreen()
+    expect(tree.getByTestId('NoActivity/loading')).toBeDefined()
+    expect(tree.queryByTestId('NoActivity/error')).toBeNull()
+    expect(tree.queryByTestId('TransactionList')).toBeNull()
+  })
+
+  it("renders an error screen if there's no cache and the query fails", async () => {
+    mockFetch.mockReject(new Error('Test error'))
+    const tree = renderScreen()
+
+    await waitFor(() => tree.getByTestId('NoActivity/error'))
+    expect(tree.queryByTestId('NoActivity/loading')).toBeNull()
+    expect(tree.queryByTestId('TransactionList')).toBeNull()
+  })
+
+  it('renders correct status for a complete transaction', async () => {
+    mockFetch.mockResponse(typedResponse({ transactions: [mockTransaction()] }))
+    const tree = renderScreen()
+
+    await waitFor(() => tree.getByTestId('TransactionList'))
+    expect(tree.getByText('feedItemReceivedInfo, {"context":"noComment"}')).toBeTruthy()
+  })
+
+  it('renders correct status for a failed transaction', async () => {
+    mockFetch.mockResponse(
+      typedResponse({ transactions: [mockTransaction({ status: TransactionStatus.Failed })] })
+    )
+
+    const tree = renderScreen()
+
+    await waitFor(() => tree.getByTestId('TransactionList'))
+    expect(tree.getByText('feedItemFailedTransaction')).toBeTruthy()
+  })
+
+  it('tries to fetch 20 transactions, unless the end is reached', async () => {
```

Comment:
Maybe this should be the name of the new test?
And there would be another one checking it fetches the first page initially?
```suggestion
  it('fetches the next page by scrolling to the end of the list', async () => {
```

I don't think we need the previous behavior of automatically loading the next page
that we added in https://github.com/valora-inc/wallet/pull/2779

Right?

Reply:
@jeanregisser
1. Sorry for the confusion! I've already renamed this and the next test in the [next PR](https://github.com/valora-inc/wallet/pull/6136/files#diff-d5f32e29af91d01b27c6e117554b1b1274175f340101fc3cc0b52ef5dd2e7d91R214)

2. Agree, will add the extra test!